packages/typespec-azure-core/lib/foundations.tsp

```diff
@@ -168,7 +168,6 @@ model InnerError {
   code?: string;
 
   @doc("Inner error.")
-  @projectedName("csharp", "InnerErrorObject")
```

Comment:
This feels like a breaking change for c-sharp, can we add this decoration in tcgc?